tools/cmake/mbed_set_post_build.cmake

```diff
@@ -87,6 +87,12 @@ function(mbed_set_post_build target)
     # The mapfile name includes the top-level target name and the
     # executable suffix for all toolchains as CMake hardcodes the name of the
     # diagnostic output file for some toolchains.
+
+    # COPY_FILE could be used with cmake >=3.21
+    file(COPY_FILE 
```

Comment:
Why using the new `COPY_FILE` rather than the `COPY` existing in 3.19?
They both seem to be very similar: https://cmake.org/cmake/help/latest/command/file.html#copy-file

Also, if my understanding is correct, they are executing at configure stage while `mbed_set_post_build` runs things after the build.
What you want here may instead be an `add_custom_command` with a regular `mv` command.

Reply:
I did not manage it with COPY, this command wants to copy a file to another directory. No renaming or mv to another name. And errors must be ignored, memap works also without the .old file.
The mv command is OS depending, so I think that’s the reason for the built in file commands.
If some other method for Os independent renaming exists, it will be fine also.

Reply:
CMake has a few built-in OS-independent commands, e.g. 
```
$ cmake -E copy <file> <destination>
```
The path to `cmake` can be obtained from a built-in variable [`CMAKE_COMMAND`](https://cmake.org/cmake/help/latest/variable/CMAKE_COMMAND.html).

Reply:
I think the cmake -E copy via add_custom_target would be the ideal solution, or ?

Reply:
I have not so much experience with cmake, so I don't know where to put this exactly. If this command can copy the .map to .map.old without throwing errors when
- the sourcefile .map does not exist
- the destfile .old already exists
then it will a solution too.

Reply:
ok, got it. It can be done with the cmake -E rename command.
rename will silently overwrite an existing destination file, only the case if source does not exist needs to be checked.
Thanks for pointing me to this possibilities.